doc/devel/development.md

```diff
@@ -0,0 +1,1932 @@
+# Development documentation
+
```

Comment:
As far as I understand, this file is a merge of several things:

1. old development.md document
2. old howto.md document
3. old usernmanual.md document
4. new generated content (not sure...)

For reviewing, if would be great if we could tell the mapping between the sections in the ToC and the 4 categories above.

Reply:
This is where the content comes from:

-   [Preface](#preface)

### doc/development.md
-   [Contributing](#contributing)
-   [Project management](#project-management)
    -   [Installing dependencies](#installing-dependencies)
    -   [Project build](#project-build)
    -   [Testing](#testing)
        -   [Test requirements](#test-requirements)
        -   [Debug Test](#debug-test)
        -   [Continuous testing](#continuous-testing)
        -   [Code Coverage](#code-coverage)
    -   [Clean](#clean)
    -   [Checking code style](#checking-code-style)
        -   [Source code style validation - ESLint](#source-code-style-validation---eslint)
        -   [Documentation Markdown validation](#documentation-markdown-validation)
        -   [Documentation Spell-checking](#documentation-spell-checking)
        -   [Prettify Code](#prettify-code)

### doc/usermanual.md
-   [Library functions and modules](#library-functions-and-modules)
    -   [Stats Registry](#stats-registry)
    -   [Alarm module](#alarm-module)
    -   [Transactions](#transactions)
    -   [Library overview](#library-overview)
    -   [Function reference](#function-reference)
        -   [Generic middlewares](#generic-middlewares)

### doc/development.md

-   [DB Models from API document](#db-models-from-api-document)
    -   [Service group model](#service-group-model)
    -   [Device model](#device-model)
-   [Data mapping plugins](#data-mapping-plugins)
    -   [Plugins usage](#plugins-usage)
    -   [Provided plugins](#provided-plugins)


### doc/howto.md

-   [Developing a new IoT Agent](#developing-a-new-iot-agent)
    -   [Protocol](#protocol)
    -   [Requirements](#requirements)
    -   [Basic IOTA](#basic-iot-agent)
    -   [IOTA With Active attributes](#iot-agent-with-active-attributes)
    -   [IOTA With Lazy attributes](#iota-with-lazy-attributes)
        -   [Previous considerations](#previous-considerations)
        -   [Implementation](#implementation)
    -   [IoT Agent in multi-thread mode](#iot-agent-in-multi-thread-mode)
    -   [Configuration management](#configuration-management)
        -   [Provisioning handlers](#provisioning-handlers)

Reply:
"Preface" is not coming from any existing document? Or it has been written from scratch?

Reply:
You are right, Preface is brand new section

Reply:
It comes from https://github.com/telefonicaid/iotagent-node-lib/blob/d90c0d7e7a82c60c9979eb709ce150bc39fc6b98/doc/howto.md?plain=1#L59

Reply:
NTC